specs/agents/metadata.md

```diff
@@ -24,42 +24,38 @@ System metadata relates to the host/container in which the service being monitor
 
 #### Hostname
 
-This hostname reported by the agent is mapped by the APM Server to the 
-[`host.hostname` ECS field](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname), which should 
-typically contain what the `hostname` command returns on the host machine. However, since we rely on this field for 
-our integration with beats data, we should attempt to follow a similar logic to the `os.Hostname()` Go API, which beats 
-relies one. While `os.Hostname()` contains some complex OS-specific logic to cover all sorts of edge cases, our 
-algorithm should be simpler. It relies on the execution of external commands with a fallback to standard environment 
-variables. Agents SHOULD implement this hostname discovery algorithm wherever possible:
+The hostname value(s) reported by the agent are mapped by APM Server to the ECS
+[`host.hostname`](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-hostname) and
+[`host.name`](https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-name) fields.
+
+Agents SHOULD return the lower-cased FQDN whenever possible, which might require a DNS query.
+
+Agents SHOULD implement this hostname discovery algorithm wherever possible:
 ```
 var hostname;
 if os == windows
-  hostname = exec "cmd /c hostname"                   // or any equivalent *
+  // https://stackoverflow.com/questions/12268885/powershell-get-fqdn-hostname
+  // https://learn.microsoft.com/en-us/dotnet/api/system.net.dns.gethostentry
+  hostname = exec "powershell.exe [System.Net.Dns]::GetHostEntry($env:computerName).HostName" // or any equivalent *
+  if (hostname == null || hostname.length == 0)
+    hostname = exec "cmd.exe /c hostname"               // or any equivalent *
   if (hostname == null || hostname.length == 0)
     hostname = env.get("COMPUTERNAME")
 else 
-  hostname = exec "uname -n"                          // or any equivalent *
-  if (hostname == null || hostname.length == 0)
-    hostname = exec "hostname"                        // or any equivalent *
+  hostname = exec "hostname -f"                         // or any equivalent *
   if (hostname == null || hostname.length == 0)
     hostname = env.get("HOSTNAME")
   if (hostname == null || hostname.length == 0)
     hostname = env.get("HOST")
```

Comment:
As a final nit, I am not implementing this fallback of using the `HOSTNAME` and `HOST` envvars in the Node.js APM agent. 

I couldn't quickly find any references for the `HOST` envvar being set by anything authoritative. The `HOSTNAME` var is set by *Bash*, which doesn't strike me as an authority we should use. FWIW, [this SO post](https://superuser.com/questions/132489/hostname-environment-variable-on-linux) points out that `HOSTNAME` is not an envvar defined by POSIX (https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html).

[This earlier spec comment](https://github.com/elastic/apm/pull/517#issuecomment-940973458) was the only discussion around having the `HOSTNAME` and `HOST` fallbacks that I could see. I don't know if it originated from what Beats were doing. My understanding is that *now* beats are using an algorithm that does *not* use these vars (https://github.com/elastic/beats/blob/c7c18b37eb34f0665f6d9a3b2c7ea4543018ba6a/libbeat/cmd/instance/beat.go#L789-L803 https://github.com/elastic/go-sysinfo/blob/9ea2eba5301c77b511c0475d7137ea3437a55a9a/providers/shared/fqdn.go#L29-L43)

Regarding current usage, it looks to me like only the Java and .NET agents are currently looking at the HOSTNAME and HOST envvars. (FWIW. the .NET agent looks at all of HOSTNAME, HOST, and COMPUTERNAME on all platforms). None of the other agents that I saw are using those envvars.

Reply:
Thanks for the review and feedback @trentm !

I agree with you that those variables might definitely not standard nor available to the runtime as they are usually set by shell and are unlikely to be set on all OSes and distributions. I am assuming (slight speculation on my end) the intent to have them was to provide a "fallback just in case", and that they might not be available most of the time.

However, I don't see any major downside to having them in the spec and using them as fallback, even if it's rarely used in practice. So for now I would suggest we keep them as-is and revisit this decision later if that proves to be an issue or a source of extra complexity.